docs/guide/CSS.md

```diff
@@ -0,0 +1,395 @@
+}
+```
+
+The type selector will also match a widget's base classes. Consequently a `Static` selector will also style the button because the `Button` Python class extends `Static`.
+
+```sass
+Static {
+  background: blue;
+  border: rounded white;
+}
+```
+
+!!! note
+
+    The fact that the type selector matches base classes is a departure from browser CSS which doesn't have the same concept.
+
+You may have noticed that the `border` rule exists in both Static and Button. When this happens, Textual will use the most recently defined sub-class within a list of bases. So Button wins over Static, and Static wins over Widget (the base class of all widgets). Hence if both rules were in a stylesheet, the buttons would be "solid blue" and not "rounded white".
+
+### ID selector
+
+Every Widget can have a single `id` attribute, which is set via the constructor. The ID should be unique to it's container.
+
+Here's an example of a widget with an ID:
+
+```python
+yield Button(id="next")
+```
+
+You can match an ID with a selector starting with a hash (`#`). Here is how you might draw a red outline around the above button:
+
+```sass
+#next {
+  outline: red;
+}
+```
+
+A Widget's `id` attribute can not be changed after the Widget has been constructed. 
+
+### Class-name selector
+
+Every widget can have a number of class names applied. The term "class" here is borrowed from web CSS, and has a different meaning to a Python class. You can think of a CSS class as a tag of sorts. Widgets with the same tag will share styles.
+
+CSS classes are set via the widgets `classes` parameter in the constructor. Here's an example:
+
+```python
+yield Button(classes="success")
+```
+
+This button will have a single class called `"success"` which we could target via CSS to make the button a particular color.
+
+You may also set multiple classes separated by spaces. For instance, here is a button with both an `error` class and a `disabled` class:
+
+```python
+yield Button(classes="error disabled")
+```
+
+To match a Widget with a given class in CSS you can precede the class name with a dot (`.`). Here's a rule with a class selector to match the `"success"` class name:
+
+```sass
+.success {
+  background: green;
+  color: white;
+}
+```
+
+!!! note
+
+    You can apply a class name to any class, which means that widgets of different types could share classes.
+
+Class name selectors may be _chained_ together by appending another full stop and class name. The selector will match a widget that has _all_ of the class names set. For instance, the following sets a red background on widgets that have both `error` _and_ `disabled` class names.
+
+```sass
+.error.disabled {
+  background: darkred;
+}
+```
+
+Unlike the `id` attribute a Widget's classes can be changed after the Widget was created. Adding and removing CSS classes is the recommended way of changing the display while your app is running. There are a few methods you can use to manage CSS classes.
```

Comment:
I feel some inconsistencies in the casing of Python class names/nouns of the things we are describing.
E.g., I would capitalised `Widget` differently:

```suggestion
Unlike the `id` attribute, a widget's classes can be changed after the widget was created. Adding and removing CSS classes is the recommended way of changing the display while your app is running. There are a few methods you can use to manage CSS classes.
```

```suggestion
Unlike the `id` attribute, classes can be changed after the Widget instance was created. Adding and removing CSS classes is the recommended way of changing the display while your app is running. There are a few methods you can use to manage CSS classes.
```

Reply:
Depends wether I'm refering to the concept of widgets or the Widget class. It's not clear sometimes!